utils/protoproxy.go

```diff
@@ -43,10 +44,11 @@ type ProtoProxy[T proto.Message] struct {
 // this should be used for updates that should be sent periodically, but does not have the urgency of immediate delivery
 // updateFn should provide computations required to generate the protobuf
 // if refreshInterval is 0, then proxy will only update on MarkDirty(true)
-func NewProtoProxy[T proto.Message](refreshInterval time.Duration, updateFn func() T) *ProtoProxy[T] {
+func NewProtoProxy[T proto.Message](refreshInterval time.Duration, updateFn func() T, equalityFn func(T, T) bool) *ProtoProxy[T] {
 	p := &ProtoProxy[T]{
 		updateChan:      make(chan struct{}, 1),
 		updateFn:        updateFn,
+		equalityFn:      equalityFn,
```

Comment:
is it meaningfully different between types? or is using [proto.Equal](https://pkg.go.dev/google.golang.org/protobuf/proto#Equal) be sufficient?

Reply:
I know we are calling this `protoProxy`, but as it is using a generic type, I just gave the flexibility to the instantiating module as it will know about the type. If you think, `proto.Equal` is fine, I can remove the function.

Reply:
oh `T` is just `proto.Message`. Sorry, ok, will remove it.